docs/sources/development/faq.md

```diff
@@ -0,0 +1,34 @@
+# Frequently Asked Questions (FAQs)
+
+Here is a list of frequently asked question, sorted by category. If you do not find an answer to your question, do not hesitate to ask it on the support forum.
+
+## Bindings
+
+### Structuring Things and Thing Types
+
+  1. _I am implementing a binding for system X. Shall I design this as one Thing or as a Bridge with multiple Things for the different functionalities?_ 
+  
+    In general, both options are valid:
+
+    1. You have one Thing which has channels for all your actors, sensors and other functions
+    2. You have one Bridge and an additional Thing for every actor and sensor and they would hold the channels
+
+    The preferred architecture is the latter, if this is feasible. This means that the physical devices should be represented by a Thing each. This only makes sense if your system allows you to identify the different physical devices at all. Especially, such an architecture is useful if you can do a discovery of new devices that could then be presented to the user/admin.
+    
+    If your system does not provide you any possibility to get hold of such information, but rather only shows you a "logical" view of it, then 1) is also a valid option to pursue.
+  
+  1. _Do I have to create XML files in `ESH-INF/thing` for all devices or is there any other option?_
+
+    No, the XML files are only one way to describe your devices. Alternatively, you can implement your own [ThingTypeProvider](https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingTypeProvider.java), through which you can provide thing descriptions in a programmatic way.
+
+  1. _For my system XY, there are so many different variants of devices. Do I really need to define a thing type for every single one of them?_
+
+    Thing types are important if you have no chance to request any structural information about the devices from your system and if you need users to manually chose a thing to add or configure (i.e. there is also no automatic discovery). The thing types that you provide will be the list the user can choose from. If your system supports auto-discovery discovery and you can also dynamically construct things (and their channels) from structural information that you can access during runtime, there is in theory no need to provide any thing type description at all.
```

Comment:
I would not say that there is "no need to provide any thing type description at all", if all things can be discovered and types can created dynamically. 

A static description of thing types has the advantage that the user knows which kind of devices are supported, no matter if he has a device at home or not. Especially for creating dynamic documentation and a list of all supported devices in Eclipse SmartHome and openHAB the static definitions are quite helpful. So for me static descriptions should always be preferred if possible. For sure, if not all devices are known at the development time, there is no other way to do it. But I think we should give the advice to describe devices with XML and only if it is not possible, the binding can provide the thing types dynamically.